src/main/java/seedu/fitnus/Parser.java

```diff
@@ -0,0 +1,204 @@
+            throw new IncompleteMealException();
+        }
+        int descriptionIndex = command.indexOf("m/") + 2;
+        int sizeIndex = command.indexOf("s/") + 2;
+        if (sizeIndex >= command.length()) {
+            throw new IncompleteMealException();
+        }
+        mealDescription = command.substring(descriptionIndex, sizeIndex - 2).trim();
+        if (mealDescription.isEmpty()) {
+            throw new IncompleteMealException();
+        }
+        if (!Meal.getNutrientDetails().containsKey(mealDescription)) {
+            throw new UnregisteredMealException();
+        }
+        mealSize = Integer.parseInt(command.substring(sizeIndex).trim());
+    }
+
+    public static void parseDrink(String command) throws IncompleteDrinkException, UnregisteredDrinkException {
+        if (!command.contains("d/") || !command.contains("s/")) {
+            throw new IncompleteDrinkException();
+        }
+        int descriptionIndex = command.indexOf("d/") + 2;
+        int sizeIndex = command.indexOf("s/") + 2;
+        if (sizeIndex >= command.length()) {
+            throw new IncompleteDrinkException();
+        }
+        drinkDescription = command.substring(descriptionIndex, sizeIndex - 2).trim();
+        if (drinkDescription.isEmpty()) {
+            throw new IncompleteDrinkException();
+        }
+        if (!Drink.getNutrientDetails().containsKey(drinkDescription)) {
+            throw new UnregisteredDrinkException();
+        }
+        drinkSize = Integer.parseInt(command.substring(sizeIndex).trim());
+    }
+
+    public static void parseWater(String command) throws IncompleteWaterException {
+        if (!command.contains("s/")) {
+            throw new IncompleteWaterException();
+        }
+        int sizeIndex = command.indexOf("s/") + 2;
+        if (sizeIndex >= command.length()) {
+            throw new IncompleteWaterException();
+        }
+        waterSize = Integer.parseInt(command.substring(sizeIndex).trim());
+    }
+
+    public static String parseInfoMeal(String command) throws UnregisteredMealException {
+        int mealIndex = 9;
+        String infoMealDescription = command.substring(mealIndex).trim();
+        if (!Meal.getNutrientDetails().containsKey(infoMealDescription)) {
+            throw new UnregisteredMealException();
+        }
+        return infoMealDescription;
+    }
+
+    public static String parseInfoDrink(String command) throws UnregisteredDrinkException {
+        int drinkIndex = 10;
+        String infoDrinkDescription = command.substring(drinkIndex).trim();
+        if (!Meal.getNutrientDetails().containsKey(infoDrinkDescription)) {
+            throw new UnregisteredDrinkException();
+        }
+        return infoDrinkDescription;
+    }
+
+    public static void parseEditMeal(String command) {
+        int mealSizePosition = command.indexOf("/");
+        editMealIndex = Integer.parseInt(command.substring(9, mealSizePosition - 2).trim()) - 1;
```

Comment:
Try to avoid magic numbers and create a constant with a meaninful name instead, e.g. offsetOne